### PR TITLE
Feature/10 create a swagger document

### PR DESCRIPTION
**Pull request by hitohata**
## Overviews of implementation

Add an OpenAPI document

## Review points

* add a document

### CODE REVIEW

File: web-api/nishiki-web-api.yaml [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+              properties:
+                name:
+                  description: user's name
+                  type: string
+      responses:
+        "202":
+          description: update user's information
+        "401":
+          description: try to change other user's information
+  /users/:userId/containers:
```
[the remark this answers is not in the document]
Comment (taniguchiiqqq) [reply web-api/nishiki-web-api.yaml:40]: The "parameters" key is needed for this doc to work on the Swagger UI.

```
      parameters:
        - name: userId
          in: path
          description: ID of user
          required: true
          schema:
            type: string
```

File: web-api/nishiki-web-api.yaml [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+                properties:
+                  containerId:
+                    type: string
+                    description: generated container ID
+                    example: containerId
+  /containers/:containerId:
+    get:
+      tags:
+        - container
+      responses:
```
Comment (taniguchiiqqq) [inline web-api/nishiki-web-api.yaml:87]: The "parameters" key is needed for this doc to work on the Swagger UI.

```
      parameters:
        - name: containerId
          in: path
          description: ID of container
          required: true
          schema:
            type: string
```

File: web-api/nishiki-web-api.yaml [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+          description: get a container
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Container"
+  /containers/:containerId/foods:
+    post:
+      tags:
+        - container
+      requestBody:
```
[the remark this answers is not in the document]
Comment (taniguchiiqqq) [reply web-api/nishiki-web-api.yaml:98]: The "parameters" key is needed for this doc to work on the Swagger UI.

```
      parameters:
        - name: containerId
          in: path
          description: ID of container
          required: true
          schema:
            type: string
```

[the remark this answers is not in the document]
Comment (taniguchiiqqq) [reply web-api/nishiki-web-api.yaml:98]: I think this endpoint requires a request body, so having a "required" key is better.

File: web-api/nishiki-web-api.yaml [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+                type: object
+                properties:
+                  foodId:
+                    type: string
+                    example: foodId
+  /containers/:containerId/foods/:foodId:
+    put:
+      tags:
+        - container
+      requestBody:
```
[the remark this answers is not in the document]
Comment (taniguchiiqqq) [reply web-api/nishiki-web-api.yaml:136]: The "parameters" key is needed for this doc to work on the Swagger UI.

```
      parameters:
        - name: containerId
          in: path
          description: ID of container
          required: true
          schema:
            type: string
        - name: foodId
          in: path
          description: ID of food
          required: true
          schema:
            type: string
```

File: web-api/nishiki-web-api.yaml [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+                  type: string
+                  format: date-time
+                  nullable: true
+                  example: 1970-01-01T00:00:00.000Z
+      responses:
+        "202":
+          description: update complete
+    delete:
+      tags:
+        - container
```
Comment (taniguchiiqqq) [inline web-api/nishiki-web-api.yaml:164]: The "parameters" key is needed for this doc to work on the Swagger UI.

```
      parameters:
        - name: containerId
          in: path
          description: ID of container
          required: true
          schema:
            type: string
        - name: foodId
          in: path
          description: ID of food
          required: true
          schema:
            type: string
```

File: web-api/nishiki-web-api.yaml [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+              schema:
+                type: object
+                properties:
+                  groupId:
+                    type: string
+                    example: group1
+  /groups/:groupId:
+    get:
+      tags:
+        - group
```
Comment (taniguchiiqqq) [inline web-api/nishiki-web-api.yaml:196]: The "parameters" key is needed for this doc to work on the Swagger UI.

```
      parameters:
        - name: grouoId
          in: path
          description: ID of group
          required: true
          schema:
            type: string
```

File: web-api/nishiki-web-api.yaml [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+        "200":
+          description: get a group information
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Group"
+  /groups/:groupId/users/:userId:
+    put:
+      tags:
+        - group
```
Comment (taniguchiiqqq) [inline web-api/nishiki-web-api.yaml:207]: The "parameters" key is needed for this doc to work on the Swagger UI.

```
      parameters:
        - name: grouoId
          in: path
          description: ID of group
          required: true
          schema:
            type: string
        - name: userId
          in: path
          description: ID of user
          required: true
          schema:
            type: string
```

File: web-api/nishiki-web-api.yaml [hunk 8]
```diff
@@ -0,0 +1,299 @@
...
+      description: add the target user ID in the path parameter
+      responses:
+        "202":
+          description: add a user to the group
+    delete:
+      tags:
+        - group
```
Comment (taniguchiiqqq) [inline web-api/nishiki-web-api.yaml:214]: The "parameters" key is needed for this doc to work on the Swagger UI.

```
      parameters:
        - name: grouoId
          in: path
          description: ID of group
          required: true
          schema:
            type: string
        - name: userId
          in: path
          description: ID of user
          required: true
          schema:
            type: string
```

**Review by taniguchiiqqq — CHANGES_REQUESTED** [review]
Thank you for creating the OpenAPI Specification definition file. I mentioned some points basically to make this doc work on the Swagger UI.

**Comment by hitohata** [comment]
@taniguchiiqqq 
Thank you for the review! I fixed it.

**Comment by taniguchiiqqq** [comment]
@hitohata 
Thank you for fixing.
Could you remove colons : from path parameters? {Path parameter's name} must have the same name as the parameters: section's -name.

**Comment by hitohata** [comment]
@taniguchiiqqq 
Sorry, I deleted the colon symbol.

**Review by taniguchiiqqq — APPROVED** [review]
Thank you!!

**Pull request merged**